8.0 prototype

foutoucour opened the pull request:
# Module Prototype

This module allows the administrator to prototype new features and export them as module. 
Functional people can prepare the job for a developer who is left with the logic to implement 
in addition to everything the prototype does not export yet.
See README.rst for more information.
# Version 0.1

All the features are not implemented but the minimal is done.
With this version you can:
- The set up of `__openerp__.py` is covered, description, maintainer, website, name, technical name...
- Views and menus can be set through odoo and gathered in prototype. The files will be automatically generated and add to the data section of the `__openerp__.py`. Be aware some advanced feature as domain or context might still missing.
- Dependencies can be set throught the Dependency page
- Custom fields can be added. A file by model will be generated with all the fields of the model. The **init**.py files are updated accordingly. Be aware that some features are not implemented yet, as the domain, the context.

**This version should be enough for:**
- set up the client module
- create a new field in a view.

**Features that you might expect in next version:**
- better generation of fields (many2many fields, domain, context)
- Generation of data and demo data files.
- Security files (rules, ir.model.access.csv)
- `__openerp__`'s description editing README.rst and index.html
- module screenshots 
# Version 0.2
- Renamed from prototype to module_prototyper as discussed in https://github.com/OCA/server-tools/issues/108
- menu in Settings that gather element used to create a prototype (menu views, views, fields)
## Known bugs
- icon file extension might be wrong. The extension is hard coded for now. It is planned add Document as dependency to handle it.

Hope it helps,
Jordi

Commit:
[ADD] Initial version of prototype
[IMP] Add wizard for API version and templates. Update translation file
[ADD] Security template files
[ADD] Filters for data and demo data

Commit:
[IMP] prototype: implementation of jinja syntax.

[IMP] prototype: Switched to v8 api.

[IMP] prototype: renamed the templates to avoid confusion with pep8 tests.

[IMP] prototype: renamed some fields in prototype.py.

[IMP] prototype: implemented the process to create the zip file to export.

[IMP] prototype: worked on the generation of __openerp__.py and __init__.py files from the prototype.

[IMP] prototype: pep8

[IMP] prototype: added more comments.

[IMP] prototype: worked on fields.

[IMP] prototype: implemented the generation of files from models/model_name.py.template and models/init.py.template

[IMP] prototype: first work on implementation of generations of views.

[IMP] prototype: improved the arch section in model_view.xml.template

[IMP] prototype: Reviewed the names of many2many from plurials to name_ids.

[IMP] prototype: added menus.

[IMP] prototype: typo in templates/8.0/views/model_view.xml.template

[IMP] prototype: improved template with adding extension to header.template to ease management of file with licence header.

[IMP] prototype: prototype.prototype demo data

[IMP] prototype: worked to get the views and menus properly generated. Added actions to model_menus.xml.template.

[IMP] prototype: should fix pep8 in model_name.py.template.

Commit:
[IMP] prototype: implementation of jinja syntax.

[IMP] prototype: Switched to v8 api.

[IMP] prototype: renamed the templates to avoid confusion with pep8 tests.

[IMP] prototype: renamed some fields in prototype.py.

[IMP] prototype: implemented the process to create the zip file to export.

[IMP] prototype: worked on the generation of __openerp__.py and __init__.py files from the prototype.

[IMP] prototype: pep8

[IMP] prototype: added more comments.

[IMP] prototype: worked on fields.

[IMP] prototype: implemented the generation of files from models/model_name.py.template and models/init.py.template

[IMP] prototype: first work on implementation of generations of views.

[IMP] prototype: improved the arch section in model_view.xml.template

[IMP] prototype: Reviewed the names of many2many from plurials to name_ids.

[IMP] prototype: added menus.

[IMP] prototype: typo in templates/8.0/views/model_view.xml.template

[IMP] prototype: improved template with adding extension to header.template to ease management of file with licence header.

[IMP] prototype: prototype.prototype demo data

[IMP] prototype: worked to get the views and menus properly generated. Added actions to model_menus.xml.template.

[IMP] prototype: should fix pep8 in model_name.py.template.

blaggacao committed:
Inline comments :: 'js':[],'css':[],'qweb':[]

I propose to add inline comments for the following reason:

General understanding is promoted when reading the source files as probablyt the main entry point for a developer.
js,css,qweb are taken from other manifests I've seen. They are explained in the link. I think completedness of the manifest scaffold also helps reduce ambiguity about it.

Lastly, I introduced newlines for better readability and grouping.

I hope these changes, well (marginally) improve the usefulness of this great module.

blaggacao committed:
Inline comments :: 'js':[],'css':[],'qweb':[]

I propose to add inline comments for the following reason:

General understanding is promoted when reading the source files as probablyt the main entry point for a developer.
js,css,qweb are taken from other manifests I've seen. They are explained in the link. I think completedness of the manifest scaffold also helps reduce ambiguity about it.

Lastly, I introduced newlines for better readability and grouping.

I hope these changes, well (marginally) improve the usefulness of this great module.

Commit:
[IMP] Module description and dependencies

[FIX] URL

[FIX] URL to issue

[FIX] Typo

[FIX] Typo

Commit:
[IMP] Module description and dependencies

[FIX] URL

[FIX] URL to issue

[FIX] Typo

[FIX] Typo

Commit:
[IMP] prototype: added helper fields and placeholders on new fields.

[IMP] prototype: no more bug when no icon, commented no implemented page in the view.

[IMP] prototype: better management of special cases.

[IMP] prototype: prototype.py removed print statement and shadowing of "fields".

[IMP] prototype -> module_prototyper

[IMP] module_prototyper: gathered menuitems that is helpfull to create a prototype under the menu 'Module Prototypes'

[IMP] module_prototyper: more comment and docstrings.

[IMP] module_prototyper: translations.

[FIX] module_prototyper: pep8

[IMP] update module_prototyper:

* pump up the version of the module to 0.3
* replace ir.ui.model by ir.ui.view in generated xml views
* improve pep8 compatibility of generation of models

[IMP] update module_prototyper: update of README with versions.

Commit:
[IMP] prototype: added helper fields and placeholders on new fields.

[IMP] prototype: no more bug when no icon, commented no implemented page in the view.

[IMP] prototype: better management of special cases.

[IMP] prototype: prototype.py removed print statement and shadowing of "fields".

[IMP] prototype -> module_prototyper

[IMP] module_prototyper: gathered menuitems that is helpfull to create a prototype under the menu 'Module Prototypes'

[IMP] module_prototyper: more comment and docstrings.

[IMP] module_prototyper: translations.

[FIX] module_prototyper: pep8

[FIX] module_prototyper: fix typo in tests.

[IMP] update module_prototyper:

* pump up the version of the module to 0.3
* replace ir.ui.model by ir.ui.view in generated xml views
* improve pep8 compatibility of generation of models

[IMP] update module_prototyper: update of README with versions.

dreispt commented on prototype/__init__.py:
```diff
@@ -0,0 +1,25 @@
+from . import (
+    models,
+    wizard
+)
```
Why use four lines when you can use 2?

max3903 commented:
#108

foutoucour commented:
test failed because of base_field_serialized. It is out of the scope of this pull request.

ehdem commented on module_prototyper/models/module_prototyper.py:
```diff
@@ -0,0 +1,358 @@
+        help=('The Module Name will be used as the displayed name of the '
+              'exported module.')
+    )
+    summary = fields.Char('Summary', required=True)
```
Add a help option

ehdem commented on module_prototyper/models/module_prototyper.py:
```diff
@@ -0,0 +1,358 @@
+              'exported module.')
+    )
+    summary = fields.Char('Summary', required=True)
+    description = fields.Text('Description', required=True)
```
Add a help option

ehdem commented on module_prototyper/models/module_prototyper.py:
```diff
@@ -0,0 +1,358 @@
+    )
+    summary = fields.Char('Summary', required=True)
+    description = fields.Text('Description', required=True)
+    author = fields.Char('Author', required=True)
```
Add a help option

ehdem commented on module_prototyper/models/module_prototyper.py:
```diff
@@ -0,0 +1,358 @@
+    summary = fields.Char('Summary', required=True)
+    description = fields.Text('Description', required=True)
+    author = fields.Char('Author', required=True)
+    maintainer = fields.Char('Maintainer')
```
Add a help option

ehdem commented on module_prototyper/models/module_prototyper.py:
```diff
@@ -0,0 +1,358 @@
+    description = fields.Text('Description', required=True)
+    author = fields.Char('Author', required=True)
+    maintainer = fields.Char('Maintainer')
+    website = fields.Char('Website')
```
Add a help option

ehdem commented on module_prototyper/models/module_prototyper.py:
```diff
@@ -0,0 +1,358 @@
+    dependency_ids = fields.Many2many(
+        'ir.module.module', 'module_prototyper_module_rel',
+        'module_prototyper_id', 'module_id',
+        'Dependencies'
```
Add a help option

ehdem commented on module_prototyper/models/module_prototyper.py:
```diff
@@ -0,0 +1,358 @@
+    field_ids = fields.Many2many(
+        'ir.model.fields', 'prototype_fields_rel',
+        'module_prototyper_id', 'field_id', 'Fields'
+    )
```
Add a help option

ehdem commented on module_prototyper/models/module_prototyper.py:
```diff
@@ -0,0 +1,358 @@
+    menu_ids = fields.Many2many(
+        'ir.ui.menu', 'prototype_menu_rel',
+        'module_prototyper_id', 'menu_id', 'Menu Items'
+    )
```
Add a help option

ehdem commented on module_prototyper/models/module_prototyper.py:
```diff
@@ -0,0 +1,358 @@
+    view_ids = fields.Many2many(
+        'ir.ui.view', 'prototype_view_rel',
+        'module_prototyper_id', 'view_id', 'Views'
+    )
```
Add a help option

ehdem commented on module_prototyper/models/module_prototyper.py:
```diff
@@ -0,0 +1,358 @@
+    group_ids = fields.Many2many(
+        'res.groups', 'prototype_groups_rel',
+        'module_prototyper_id', 'group_id', 'Groups'
+    )
```
Add a help option

ehdem commented on module_prototyper/models/module_prototyper.py:
```diff
@@ -0,0 +1,358 @@
+        'ir.model.access', 'prototype_rights_rel',
+        'module_prototyper_id', 'right_id',
+        'Access Rights'
+    )
```
Add a help option

ehdem commented on module_prototyper/models/module_prototyper.py:
```diff
@@ -0,0 +1,358 @@
+    rule_ids = fields.Many2many(
+        'ir.rule', 'prototype_rule_rel',
+        'module_prototyper_id', 'rule_id', 'Record Rules'
+    )
```
Add a help option

ehdem commented on module_prototyper/tests/__init__.py:
```diff
@@ -0,0 +1,28 @@
+##############################################################################
+from . import test_prototype_module_export
+from . import test_prototype
+
```
Better to do:

```
from . import (
    test_prototype_module_export,
    test_prototype,
)
```

ehdem commented on module_prototyper/wizard/module_prototyper_module_export.py:
```diff
@@ -0,0 +1,121 @@
+    api_version = fields.Selection(
+        [
+            ('8.0', '8.0'),
+            # ('7.0', '7.0')
```
remove this comment

Commit:
[IMP] Add help

[FIX] license field
[IMP] Add template README.rst as default description

[FIX] PEP8 errors

[IMP] Take reviews into account

Commit:
[IMP] Add help

[FIX] license field
[IMP] Add template README.rst as default description

[FIX] PEP8 errors

[IMP] Take reviews into account

max3903 commented:
Travis is complaining about a Flake8 configuration:
`UnboundLocalError: local variable 'local_dir' referenced before assignment`

No PEP8 errors for this module so
:+1:

Commit:
[IMP] module_prototyper: add placeholders to fields.

Commit:
[IMP] module_prototyper: add placeholders to fields.

Commit:
[FIX] E402 module level import not at top of file

Commit:
[FIX] E402 module level import not at top of file

legalsylvain commented on module_prototyper/models/module_prototyper.py:
```diff
@@ -0,0 +1,456 @@
+              'install, configure and use it, the roadmap or known issues.'
+              'The description will be exported in README.rst'),
+        default="""
+Module name
```
Hi, 
This text is very huge.
Maybe it could be cleaner to move the whole text into a separated file, using a default function to get it ?

jgrandguillaume commented:
Hi,

Thanks a lot for the contribs ! I've tested it and it's working fine. I'm think that this module may come a very good way to export manual configuration like the old "base_recorder" module.

:+1: for me !

Regards,

Joël

ehdem committed:
[UPD] Add a new file for default value decription field

ehdem committed:
[UPD] Add a new file for default value decription field

max3903 commented:
@blaggacao Can you sign and send your CLA ?

http://odoo-community.org/page/cla

Thank you!

max3903 commented:
Few issues I noted while using it:
- [x] The name of the module is not set in the "name" key of the module manifest. It is currently using the technical name.
- [x] In the manifest, the category is set in the language of the user. It should always be in english.
- [x] Does not support multi lines notes on fields. Only the first line is commented, resulting in a syntax error
- [x] Description template should not have spaces at the beginning of each line

Commit:
fix template description being indented

[FIX] Use human name and english category name for manifest

[IMP] multi line notes, unprefix names, pep8 py files

[IMP] whitespace cleanup in templates

[FIX] base models get _name, custom get _inherit

flake fix

add missing spaces

prevent crash on menus with no action

switch GPL-2 to LGPL-3

generate license headers with license name

fix pep/flake errors in generated files

add m2m and relation field options in fields

Commit:
fix template description being indented

[FIX] Use human name and english category name for manifest

[IMP] multi line notes, unprefix names, pep8 py files

[IMP] whitespace cleanup in templates

[FIX] base models get _name, custom get _inherit

add missing spaces

veloutin commented:
@max3903 fixed!
Also, as a bonus, the python files generated are more pep8-compliant.

gurneyalex commented on module_prototyper/models/module_prototyper.py:
```diff
@@ -0,0 +1,450 @@
+    _description = "Module Prototyper"
+
+    license = fields.Selection(
+        [('GPL-2', 'GPL Version 2'),
```
I'd remove GPL-2 strict from the list of proposed licenses, as it is incompatible with AGPL-3 used by the core and most OCA addons. Maybe add LGPL-3 instead.

gurneyalex commented on module_prototyper/models/module_prototyper.py:
```diff
@@ -0,0 +1,450 @@
+    description = fields.Text(
+        'Description',
+        required=True,
+        help=('Enter the description of your module, what it does, how to'
```
missing space before closing quote

gurneyalex commented on module_prototyper/models/module_prototyper.py:
```diff
@@ -0,0 +1,450 @@
+        'Description',
+        required=True,
+        help=('Enter the description of your module, what it does, how to'
+              'install, configure and use it, the roadmap or known issues.'
```
missing space before closing quote

gurneyalex commented on module_prototyper/models/module_prototyper.py:
```diff
@@ -0,0 +1,450 @@
+    author = fields.Char('Author', required=True, help=('Enter your name'))
+    maintainer = fields.Char(
+        'Maintainer',
+        help=('Enter the name of the person or organization who will'
```
missing space before closing quote

gurneyalex commented on module_prototyper/models/module_prototyper.py:
```diff
@@ -0,0 +1,450 @@
+        'ir.module.module', 'module_prototyper_module_rel',
+        'module_prototyper_id', 'module_id',
+        'Dependencies',
+        help=('Enter the list of required modules that need to be installed'
```
missing space before closing quote

gurneyalex commented on module_prototyper/models/module_prototyper.py:
```diff
@@ -0,0 +1,450 @@
+        'ir.model.fields', 'prototype_fields_rel',
+        'module_prototyper_id', 'field_id', 'Fields',
+        help=('Enter the list of fields that you have created or modified'
+              'and want to export in this module. New models will be'
```
missing spaces before closing quotes

gurneyalex commented on module_prototyper/models/module_prototyper.py:
```diff
@@ -0,0 +1,450 @@
+        'ir.ui.menu', 'prototype_menu_rel',
+        'module_prototyper_id', 'menu_id', 'Menu Items',
+        help=('Enter the list of menu items that you have created and want'
+              'to export in this module. Related windows actions will be'
```
missing spaces before closing quotes

gurneyalex commented on module_prototyper/models/module_prototyper.py:
```diff
@@ -0,0 +1,450 @@
+    group_ids = fields.Many2many(
+        'res.groups', 'prototype_groups_rel',
+        'module_prototyper_id', 'group_id', 'Groups',
+        help=('Enter the list of groups that you have created and want to'
```
missing space before closing quote

gurneyalex commented on module_prototyper/models/module_prototyper.py:
```diff
@@ -0,0 +1,450 @@
+    rule_ids = fields.Many2many(
+        'ir.rule', 'prototype_rule_rel',
+        'module_prototyper_id', 'rule_id', 'Record Rules',
+        help=('Enter the list of record rules that you have created and'
```
missing space before closing quote

gurneyalex commented:
once the missing spaces in the help strings are added, :+1:

max3903 commented:
When I export, I get 

```
Traceback (most recent call last):
  File "/home/max/odoo/prototype/git/odoo/openerp/http.py", line 530, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/home/max/odoo/prototype/git/odoo/openerp/http.py", line 567, in dispatch
    result = self._call_function(**self.params)
  File "/home/max/odoo/prototype/git/odoo/openerp/http.py", line 303, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/home/max/odoo/prototype/git/odoo/openerp/service/model.py", line 113, in wrapper
    return f(dbname, *args, **kwargs)
  File "/home/max/odoo/prototype/git/odoo/openerp/http.py", line 300, in checked_call
    return self.endpoint(*a, **kw)
  File "/home/max/odoo/prototype/git/odoo/openerp/http.py", line 796, in __call__
    return self.method(*args, **kw)
  File "/home/max/odoo/prototype/git/odoo/openerp/http.py", line 396, in response_wrap
    response = f(*args, **kw)
  File "/home/max/odoo/prototype/git/odoo/addons/web/controllers/main.py", line 940, in call_button
    action = self._call_kw(model, method, args, {})
  File "/home/max/odoo/prototype/git/odoo/addons/web/controllers/main.py", line 928, in _call_kw
    return getattr(request.registry.get(model), method)(request.cr, request.uid, *args, **kwargs)
  File "/home/max/odoo/prototype/git/odoo/openerp/api.py", line 241, in wrapper
    return old_api(self, *args, **kwargs)
  File "/home/max/odoo/prototype/git/odoo/openerp/api.py", line 336, in old_api
    result = method(recs, *args, **kwargs)
  File "/home/max/odoo/prototype/git/server-tools/module_prototyper/wizard/module_prototyper_module_export.py", line 82, in action_export
    zip_details = self.zip_files(prototype.generate_files())
  File "/home/max/odoo/prototype/git/odoo/openerp/api.py", line 239, in wrapper
    return new_api(self, *args, **kwargs)
  File "/home/max/odoo/prototype/git/server-tools/module_prototyper/models/module_prototyper.py", line 227, in generate_files
    file_details.extend(self.generate_menus_details())
  File "/home/max/odoo/prototype/git/odoo/openerp/api.py", line 239, in wrapper
    return new_api(self, *args, **kwargs)
  File "/home/max/odoo/prototype/git/server-tools/module_prototyper/models/module_prototyper.py", line 346, in generate_menus_details
    relations.setdefault(menu.action.res_model, []).append(menu)
AttributeError: 'bool' object has no attribute 'res_model'
```

One of my menu item has no action. It is just a menu item containing submenu items.

Commit:
[IMP] Add license field in the view

[FIX] Typos

[IMP] application = True in module manifest

Commit:
empty selection in m2m field template

[FIX] other licenses, return lines as well

[FIX] license not shown in __oe__

[FIX] unprefix more names, try to get _name/_inherit right [IMP] group by module in zip

[FIX] fix category and summary being on same line

[FIX] fix export test

[IMP] add tabs for reports/security/workflow/data + partial data/demo generation

unprefix model names in __init__

[FIX] fix data file names in __openerp__.py

[IMP] move Data&Demo after Interface in view

[FIX] unprefix view file names

[IMP] remove prefixes from field attrs in views

[FIX] encode files in zip to utf-8, remove trailing comma in menu groups

remove unused variable in tests

remove AGPL3 or later from license choices: not in base module choices

veloutin commented:
Rebased on 8.0

bwrsandman commented on module_prototyper/tests/test_prototype_module_export.py:
```diff
@@ -0,0 +1,83 @@
+        """Test the method return of the method that generate the zip file."""
+        file_details = (
+            ('test.txt', 'generated'),
+        )
```
Nothing is done with this variable. Maybe the test is not complete?

bwrsandman commented:
Why does it generate fields with `_()` for string?
I also noticed that it exports ir.ui.view types... this is a deprecated field in the xml at least.

max3903 commented:
I used the prototype to make https://github.com/OCA/management-system/pull/78 and I get the following error message when I update the module list:

```
The value "AGPL-3 or any later version" for the field "ir_module_module.license" is not in the selection 
```

veloutin commented:
@max3903, I removed it from the choices, as it is effectively not in the base module license choices.

max3903 commented:
Is is the same with 'LGPL-3 or later version' ?

andreparames commented:
@max3903, yes, Odoo doesn't have LGPL at all in the license field selection. I've made a PR to add it: https://github.com/odoo/odoo/pull/6914

Commit:
[FIX] Default license

[IMP] Add OCA as an author

[IMP] Add Bug Tracker section in description

[IMP] Update description template

[ADD] Initial version of prototype

[IMP] Add wizard for API version and templates. Update translation file

[ADD] Security template files

[ADD] Filters for data and demo data

max3903 commented:
Travis and Runbot errors are due to other modules.

OSguard commented:
rebase don't help?

OSguard commented on module_prototyper/__openerp__.py:
```diff
@@ -0,0 +1,46 @@
+{
+    'name': 'Module Prototyper',
+    'version': '0.3',
+    'author': 'Savoir-faire Linux',
```
add OCA as author

OSguard commented on module_prototyper/models/default_description.py:
```diff
@@ -0,0 +1,90 @@
+======================
+
+ * ...
+
```
add Bug Tracker section from
https://raw.githubusercontent.com/OCA/maintainer-tools/master/template/module/README.rst

OSguard commented:
:+1: (review, no test)

OSguard commented:
@gurneyalex how we can fix the runbot problem?

yvaucher commented on module_prototyper/templates/8.0/models/model_name.py.template:
```diff
@@ -0,0 +1,68 @@
+{% block body %}
+
+from openerp import models, fields
+from openerp.tools.translate import _
```
You can import `_` from openerp

gurneyalex replied:
/me frowns on transparent imports... but then...

bwrsandman replied:
It's idiomatic of v8 to have all the usual imports on the same line from openerp.
Personally, I find it much nicer to read.

yvaucher commented on module_prototyper/models/ir_model_fields.py:
```diff
@@ -0,0 +1,49 @@
+##############################################################################
+
+from openerp import fields, models
+from openerp.tools.translate import _
```
You can import `_` from openerp

yvaucher commented on module_prototyper/models/module_prototyper.py:
```diff
@@ -0,0 +1,556 @@
+    )
+    version = fields.Char(
+        'Version',
+        size=3,
```
Shouldn't this be bigger? In fact you can write up to odoo.odoo_sub.major.minor.fix => which could be 10.2.1.4.24

max3903 replied:
This idea is to have two digits only: major.minor. 

If migration is needed, change major, otherwise increment minor.

pedrobaeza replied:
But you can have 3.10 version

yvaucher commented on module_prototyper/models/module_prototyper.py:
```diff
@@ -0,0 +1,556 @@
+        'Version',
+        size=3,
+        default='0.1',
+        help=('Enter the version of your module with 2 digits')
```
Why this limitation?

yvaucher commented on module_prototyper/wizard/module_prototyper_module_export.py:
```diff
@@ -0,0 +1,132 @@
+        default='choose'
+    )
+
+    @api.model
```
Why not using @api.multi?

yvaucher commented on module_prototyper/wizard/module_prototyper_module_export.py:
```diff
@@ -0,0 +1,132 @@
+
+        # getting the prototype of the wizard
+        prototypes = self.env[active_model].browse(
+            [self._context.get('active_id')]
```
Why not naming this variable prototype? browsing from active_id should return only a record set of size 1.

Plus, you can pass a single id instead of a list.

yvaucher commented on module_prototyper/i18n/fr.po:
```diff
@@ -0,0 +1,338 @@
+#: help:module_prototyper,data_ids:0
+msgid "The records matching the filters will be added as data."
+msgstr ""
+"Les enregistrements correspondant aux critères des filtres seront exportées "
```
s/correspondant/correspondants

yvaucher replied:
s/exportées/exportés (ce sont les enregistrements qui sont exportés)

yvaucher commented on module_prototyper/i18n/fr.po:
```diff
@@ -0,0 +1,338 @@
+#: help:module_prototyper,demo_ids:0
+msgid "The records matching the filters will be added as demo data."
+msgstr ""
+"Les enregistrements correspondant aux critères des filtres seront exportées "
```
s/correspondant/correspondants

yvaucher replied:
s/exportées/exportés

blaggacao commented:
Using it, LGTM (using) for Version 0.2 :+1:

OSguard commented:
@blaggacao thank you for your support. But please can you test & review the version of this pull request.
Talking about other version confuses more than helps the community in this pull request. We need to focus what is here the point and this is version 0.3 to get it submitted to OCA.
Kind Regards.

blaggacao commented:
@OSguard I think I'm already confused and I'm sorry for that ;) As I'm not confident with the review process, I might want to refer to https://en.wikipedia.org/wiki/Wikipedia:DNB, not ratified within OCA but valuable...
I draw on what is outlined here: https://github.com/OCA/server-tools/pull/107#issue-53123593
== OT ==
Therefore, can or should any progress not be shown or presented in a more transparent fashion in order not to scare away people like me? Or is it as you say, that in the long run, those kind of contributions do not tend to be helpful?

I was asked to leave a comment [here](https://github.com/OCA/server-tools/issues/104#issuecomment-112080687) and it is really best effort, I (or half-involved people like me) might probably tend to refuse to go beyond.
== END ==

My version is:

```
commit fe6b7862e08772e863f4cdf3725de79d96c88a5a
Author: Maxime Chambreuil <maxime.chambreuil@savoirfairelinux.com>
Date:   Fri Jun 12 17:10:21 2015 -0400

    [FIX] Default license
```

I'm using it productively so I'm a bit reluctant to update just now. But in the end, I can see your very valuable point. I hope with this commit reference it is valued anyhow as a general and summatory user feedback. Thanks!

blaggacao commented:
@OSguard (off-topic) you asked @gurneyalex about the runbot external effects and how to solve it. I want to give a suggestion:
New includes could be done against a specific branch of a "--review" repository, then all sort of checks and verifications would be relevant and we would enjoy generally less clutter. Once the review is completed, it can be merged with some of those magic merges about "how to include a subdirectory into a repository from another repository" (`git read-tree` and `git filter-branch`)

http://bneijt.nl/blog/post/merge-a-subdirectory-of-another-repository-with-git/

To cover for a optimum solution:
Make a github.com/OCA-Dev Github user, which hosts reporitories on a "per module" basis (grouping together modules that are technical *and* semantical dependencies), then achieve the convenience-grouping (which imho is confusing) of github.com/OCA with subtree-merges
https://help.github.com/articles/about-git-subtree-merges/

max3903 commented:
@blaggacao Please stop confusing this PR and use mailing list for that. As you said it is off-topic, so keep it off this topic ;)

gdgellatly commented:
@OSguard @max3903 If you could review this we could fix the failing test to make things green #194

Commit:
[DEL] cleanup
[FIX] PEP8 errors
[FIX] Get default description
[IMP] README.rst

[IMP] Provide Jinja2 requirement
[IMP] Add license badge in description
[FIX] Formatting

[FIX] Version and typos

yvaucher commented:
Still some typo in fr.po and few comments to address.

yvaucher commented on module_prototyper/data/README.rst:
```diff
@@ -0,0 +1,78 @@
+   :alt: License: AGPL-3
+
+===========
+Module name
```
Replace with placeholder {module_title} ?

max3903 replied:
This file is loaded as a placeholder for the description. It is not a template.

yvaucher commented on module_prototyper/__openerp__.py:
```diff
@@ -0,0 +1,46 @@
+
+{
+    'name': 'Module Prototyper',
+    'version': '0.3',
```
Full version ? 8.0.0.3 ?

mmalorni commented:
:+1:

ghost commented:
:+1:

pedrobaeza commented:
@max3903, please squash the commits a bit (joining subsequent commits from the the same author), because 138 commits is a lot, and then I'll merge.

blaggacao commented:
OT: Where to pul request correctly our development about toposorting data files AND field dependencies... ? @kurkop

max3903 commented:
@pedrobaeza Rebase done. Thanks for merging!

pedrobaeza commented on module_prototyper/README.rst:
```diff
@@ -0,0 +1,123 @@
+* customize your instance by adding fields and creating inherited views
+* create your menu items and their window actions
+* prepare your data and demo data by creating filters
+* create your own groups with access rights and record rules
```
These 2 steps are not yet covered by the module, so you should remove them from the usage until the module get support for them.

max3903 replied:
Yes, they are covered, a user can perform those steps.

Data, demo data and security are not exported yet, that's why they appear in the roadmap section.

pedrobaeza commented on module_prototyper/README.rst:
```diff
@@ -0,0 +1,123 @@
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/server-tools/issues>`_.
+In case of trouble, please check there if your issue has already been reported.
+If you spotted it first, help us smashing it by providing a detailed and welcomed feedback
+`here <https://github.com/OCA/server-tools/issues/new?body=module:%20module_prototyper%0Aversion:%200.3%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
```
Version should be 8.0 (Odoo version), not 0.3 (module version).

pedrobaeza commented on module_prototyper/__openerp__.py:
```diff
@@ -0,0 +1,46 @@
+
+{
+    'name': 'Module Prototyper',
+    'version': '8.0.0.0.3',
```
Version should be 8.0.0.3.0 instead, because all changes done requires a module update, so you should raise y version from x.y.z

pedrobaeza commented on module_prototyper/models/module_prototyper.py:
```diff
@@ -0,0 +1,566 @@
+    version = fields.Char(
+        'Version',
+        size=9,
+        default='8.0.0.0.1',
```
I would put as default value 8.0.1.0.0

pedrobaeza commented:
@max3903, please attend remarks from @yvaucher and me and then I merge.

Commit:
[FIX] Review comments

max3903 commented:
@pedrobaeza I think @yvaucher's comments have all been answered.
